src/System.Reflection/tests/TypeDerivedTests.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Reflection.Tests
+{
+    public class TypeDerivedTests
+    {
+        [Fact]
+        public void IsAssignableFrom_NullUnderlyingSystemType()
+        {
+            Type compareType = typeof(TypeDerivedTests);
+            var testType = new MockDerivedType();
+            Assert.Null(testType.UnderlyingSystemType);
```

Comment:
I think this can be added as more `InlineData` to the existing IsAssignableFrom test in TypeInfoTests.cs. I do not think we need a special test for this.

Reply:
The IsAssignableFrom test in TypeInfoTests specifically uses `TypeInfo` (e.g. type.GetTypeInfo()) so not a fit for this.

Reply:
The `TypeInfo` tests are really testing `Type`. They were written in the netstandard1.x days when `Type` had very few methods, and all interesting methods were on `TypeInfo`. It is not case anymore.

It is fine to keep it here if you prefer.